src/kernel/filesystem/fat32.zig

```diff
@@ -1195,16 +1216,56 @@ pub fn Fat32FS(comptime StreamType: type) type {
         /// Return: *vfs.Node
         ///     The VFS Node for the opened file/folder.
         ///
-        /// Error: Allocator.Error || ReadError || SeekError || vfs.Error
+        /// Error: Allocator.Error || vfs.Error
         ///     Allocator.Error           - Not enough memory for allocating memory
-        ///     ReadError                 - Errors when reading the stream.
-        ///     SeekError                 - Errors when seeking the stream.
         ///     vfs.Error.NoSuchFileOrDir - Error if the file/folder doesn't exist.
+        ///     vfs.Error.Unexpected      - An error occurred whilst reading the file system, this
+        ///                                 can be caused by a parsing error or errors on reading
+        ///                                 or seeking the underlying stream. If this occurs, then
+        ///                                 the real error is printed using `log.err`.
         ///
-        fn openImpl(fs: *const vfs.FileSystem, dir: *const vfs.DirNode, name: []const u8) (Allocator.Error || ReadError || SeekError || vfs.Error)!*vfs.Node {
+        fn openImpl(fs: *const vfs.FileSystem, dir: *const vfs.DirNode, name: []const u8) (Allocator.Error || vfs.Error)!*vfs.Node {
             const self = @fieldParentPtr(Fat32Self, "instance", fs.instance);
+            // TODO: Cache the files in this dir, so when opening, don't have to iterator the directory every time
```

Comment:
:+1: We should definitely add some fs caching at some point as operations will be slow otherwise